neutrons/Shiver
# Implement histograming

This PR introduces the following changes
-------------------------------------------------------

- add implementation of the histograming by connecting the histogram button with backend MakeSlice algorithm
- fix an issue related to auto closing by `qbot`
- add normalization data for testing (logs removed to reduce size)

Testing instructions
-------------------------

- clone the branch and perform editable install in an environment with `mantidworkbench`.
- start `Shiver` with
  ```python
  from shiver import Shiver
  s=Shiver()
  s.show()
  ```
- Test 1: click Histogram button and an error message should appear
![image](https://user-images.githubusercontent.com/5064852/229600992-c78dc768-765d-4008-80f4-4120a17c5511.png)
- Test 2: load MDE data, and config the parameters as below, then hit the Histogram button 
![image](https://user-images.githubusercontent.com/5064852/229601294-d388decf-e79b-4791-9a9f-9bad758fe25d.png)
  - verify the history of `Plot_1` to make sure nothing is specified in `SolidAngleWorkspace`
  ![image](https://user-images.githubusercontent.com/5064852/229601529-59dc514f-eaa5-4c52-b4f1-04ebf15fc929.png)
- Test 3: now load the normalization workspace, then
  - click `TiZr` to select it 
  - change the name to `Plot 2`
  - hit the Histogram button
  ![image](https://user-images.githubusercontent.com/5064852/229602009-7e5c9faf-a942-4494-9da9-ce0ac3cdc924.png)
  - verify that `TiZr` is specified in the history of `Plot_2`
  ![image](https://user-images.githubusercontent.com/5064852/229602283-d1b2ef71-9f6c-452c-97fb-f88fb6c74b62.png)


> IBM item: [590](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=590)
